applications/conceptual-model-editor/src/dialog/selection/filter-selection-dialog-controller.ts

```diff
@@ -0,0 +1,85 @@
+import { useMemo } from "react";
+import { type DialogProps } from "../dialog-api";
+import { Selections, SelectionsWithIdInfo, SelectionFilter } from "../../action/filter-selection-action";
+
+
+const SELECTION_FILTER_TO_CHECKBOX_NAME_MAP: Record<SelectionFilter, string> = {
+  "NORMAL-CLASS": "Include non-profile classes in result",
+  "PROFILE-CLASS": "Include class profiles in result",
+  "RELATIONSHIP": "Include normal relationships (associations)",
+  "RELATIONSHIP-PROFILE": "Include profiled relationships",
+  "GENERALIZATION": "Include generalizations",
+};
+
+
+type SelectionFilterCheckboxData = {
+  checked: boolean;
+  checkboxText: string;
+  checkboxTooltip: string;
+  selectionFilter: SelectionFilter;
+};
+
+export interface SelectionFilterState {
+  selections: SelectionsWithIdInfo,
+  setSelectionsInDiagram: (newSelections: Selections) => void;
+  selectionFilters: SelectionFilterCheckboxData[];
+};
+
+/**
+ * Creates element of type {@link SelectionFilterCheckboxData} from given arguments and puts it at the end of {@link checkboxStates} parameter.
+ * @returns The created element
+ */
+const createSelectionFilterCheckboxDataAndSaveIt = (
+  checkboxStates: SelectionFilterCheckboxData[],
+  selectionFilter: SelectionFilter
+): SelectionFilterCheckboxData => {
+  const checkboxText = SELECTION_FILTER_TO_CHECKBOX_NAME_MAP[selectionFilter];
+  const filterData = {
+      checked: true,
+      checkboxText,
+      checkboxTooltip: "",
+      selectionFilter,
+  };
+
+  checkboxStates.push(filterData);
+  return filterData;
+};
+
+export function createFilterSelectionState(
+  selections: SelectionsWithIdInfo,
+  setSelectionsInDiagram: (newSelections: Selections) => void
+): SelectionFilterState {
+  const filters: SelectionFilterCheckboxData[] = [];
+  createSelectionFilterCheckboxDataAndSaveIt(filters, "NORMAL-CLASS");
```

Comment:
Same as before seems like you just crete a big object, I would create it using a single function.
No need to make it look like something is hapenning by calling a function.

Reply:
Fixed in a same way, maybe still not ideal